asf_heat_pump_suitability/pipeline/transform/non_residential_entities.py

```
@@ -0,0 +1,77 @@
+import geopandas as gpd
+import pandas as pd
+
+# TODO these may need refinement for large cities where overlap with residential is possible
+NO_RESIDENTIAL_OVERLAP_BUILDING_TYPES = [
+    "Port Consisting of Docks and Nautical Berthing",
+    "Fire Station",
+    "Hospital",
+    "Non State Secondary Education",
+    "Secondary Education",
+    "Higher or University Education",
+    "Primary Education",
+    "Place Of Worship",
+    "Medical Care Accommodation",
+    "Museum",
+    "Special Needs Education",
+    "Further Education",
+    "Non State Primary Education",
+    "Coach Station",
+    "Police Station",
+    "Sports And Leisure Centre",
+    "Vehicular Ferry Terminal",
+    "Hospice",
+    "Bus Station",
+    "Road User Services",
+    "Passenger Ferry Terminal",
+]
+
+
+def transform_gdf_non_residential_buildings(
+    important_building_gdf: gpd.GeoDataFrame,
+    railway_station_gdf: gpd.GeoDataFrame,
+    building_gdf: gpd.GeoDataFrame,
+) -> gpd.GeoDataFrame:
+    """
+    Transform important buildings and railway data to create a dataframe of polygons representing buildings
+    which are unlikely to contain residential properties, e.g. hospitals, train stations, museums etc.
+
+    Args:
+        important_building_gdf (gpd.GeoDataFrame): OS OpenMap Local important building footprints in area of interest
+        railway_station_gdf (gpd.GeoDataFrame): OS OpenMap Local railway station point geometries in area of interest
+        building_gdf (gpd.GeoDataFrame): all building footprints in area of interest
+
+    Returns:
+        gpd.GeoDataFrame: geometries of buildings which are unlikely to contain residential properties
+    """
+    print("Creating non-residential buildings dataset...")
+    # Assert all gdfs have the same CRS
+    assert (
+        len({important_building_gdf.crs, railway_station_gdf.crs, building_gdf.crs})
+        == 1
+    )
```

should we raise an exception here?

---

I chose not to because if the assertion fails it will raise an error anyway and we can see where it failed.